In [4]:
import pandas as pd


load the csv files (the counts 0)

In [5]:
csv_files = [
    'counts_1_100.csv',
    'counts_1_313.csv',
    'counts_1_364.csv',
    'counts_1_880.csv',
    'counts_1_982.csv'
]


turn it into a datafram

In [6]:
dfs = [pd.read_csv(file, index_col=0).rename_axis('cell_name').reset_index() for file in csv_files]

combine all the matrix into one and saving it as a csv file

In [7]:
combined_df = pd.concat(dfs, ignore_index = True, sort = False).fillna(0)

combined_df.to_csv('combined_counts_1.csv', index = False)

In [8]:
df100 = pd.read_csv('counts_1_100.csv', index_col='Unnamed: 0')
df100

ACTA2  ACTG2  ACTR3B  ADGRL4  AGR2  AHR  AIF1  AKT1  ALDH1A3  \
X100.3.001       0      0       1       0     0    0     0     0        0   
X100.13.001      0      0       0       0     0    0     0     0        0   
X100.21.001      0      0       0       0     0    0     0     0        0   
X100.28.001      0      0       0       0     0    0     0     0        0   
X100.31.001      0      0       0       0     0    0     0     0        0   
...            ...    ...     ...     ...   ...  ...   ...   ...      ...   
X100.52.076      0      0       0       0     0    0     0     0        0   
X100.56.076      0      0       0       0     0    0     0     0        0   
X100.57.076      0      0       0       0     0    0     0     0        0   
X100.59.076      0      0       0       0     0    0     0     1        0   
X100.69.076      0      0       0       0     0    0     0     0        0   

             ANLN  ...  TTC6  TTYH1  TYMS  UBE2C  UBE2T  VIM  XBP1  XCL1  \
X100.3.001      1  ...     0      0     0      0      0    0     0     0   
X100.13.001     0  ...     0      0     0      0      0    0     0     0   
X100.21.001     0  ...     0      0     0      0      0    0     0     0   
X100.28.001     0  ...     0      0     0      0      0    0     0     0   
X100.31.001     0  ...     0      0     0      0      0    0     0     0   
...           ...  ...   ...    ...   ...    ...    ...  ...   ...   ...   
X100.52.076     0  ...     0      0     0      0      0    0     0     0   
X100.56.076     0  ...     0      0     0      0      0    0     0     0   
X100.57.076     0  ...     0      1     0      0      0    0     0     0   
X100.59.076     0  ...     0      0     0      0      0    0     0     0   
X100.69.076     0  ...     0      0     0      0      0    0     0     0   

             ZEB1  ZNF571  
X100.3.001      0       1  
X100.13.001     0       0  
X100.21.001     0       0  
X100.28.001     0       0  
X100.31.001     0       0  
...           ...     ...  
X100.52.076     0       0  
X100.56.076     0       0  
X100.57.076     0       0  
X100.59.076     0       0  
X100.69.076     0       0  

[1545 rows x 297 columns]

In [9]:
df313 = pd.read_csv('counts_1_313.csv', index_col='Unnamed: 0')
df313

ACTA2  ACTG2  ACTR3B  ADGRL4  AGR2  AHR  AIF1  AKT1  ALDH1A3  \
X313.18.0         0      0       0       0     0    0     0     0        0   
X313.19.0         0      0       0       0     0    0     0     0        0   
X313.22.0         0      0       0       0     1    0     0     0        0   
X313.30.0         0      0       0       0     0    0     0     0        0   
X313.1.001        0      0       0       0     0    0     0     0        0   
...             ...    ...     ...     ...   ...  ...   ...   ...      ...   
X313.85.107       0      0       1       0     0    0     0     0        0   
X313.86.107       0      0       0       0     0    0     0     0        0   
X313.103.107      0      0       0       0     0    0     0     0        0   
X313.105.107      0      0       0       0     0    0     0     0        0   
X313.110.107      1      0       0       0     0    1     1     0        1   

              ANLN  ...  TTC6  TTYH1  TYMS  UBE2C  UBE2T  VIM  XBP1  XCL1  \
X313.18.0        0  ...     0      0     0      0      0    0     0     0   
X313.19.0        0  ...     0      0     0      0      1    0     0     0   
X313.22.0        1  ...     0      0     0      0      0    0     0     0   
X313.30.0        0  ...     0      0     0      1      1    0     0     0   
X313.1.001       0  ...     0      0     0      0      0    0     0     0   
...            ...  ...   ...    ...   ...    ...    ...  ...   ...   ...   
X313.85.107      0  ...     0      0     0      0      0    0     0     0   
X313.86.107      0  ...     0      0     0      0      0    0     0     0   
X313.103.107     0  ...     0      0     0      0      0    1     0     0   
X313.105.107     2  ...     0      0     0      0      0    0     0     0   
X313.110.107     0  ...     0      0     0      0      0    0     0     0   

              ZEB1  ZNF571  
X313.18.0        0       0  
X313.19.0        0       0  
X313.22.0        0       0  
X313.30.0        0       1  
X313.1.001       0       0  
...            ...     ...  
X313.85.107      0       0  
X313.86.107      0       0  
X313.103.107     0       0  
X313.105.107     1       0  
X313.110.107     0       0  

[645 rows x 291 columns]

In [10]:
df364 = pd.read_csv('counts_1_364.csv', index_col='Unnamed: 0')
df364

ACTA2  ACTG2  ACTR3B  ADGRL4  AHR  AIF1  AKT1  ALB  ALDH1A3  \
X364.9.0         0      0       0       0    0     0     0    0        0   
X364.11.0        0      0       0       0    0     0     0    0        0   
X364.13.0        0      0       0       1    0     0     0    0        0   
X364.10.001      0      0       0       0    0     0     0    0        0   
X364.12.001      0      0       0       0    0     0     0    0        0   
...            ...    ...     ...     ...  ...   ...   ...  ...      ...   
X364.79.069      0      0       0       0    0     0     0    0        0   
X364.4.071       0      0       0       0    0     0     0    0        0   
X364.8.071       0      0       0       0    0     0     0    0        0   
X364.2.074       0      0       1       0    0     0     0    0        0   
X364.9.074       0      0       0       0    0     0     0    0        0   

             ANLN  ...  TTC6  TTYH1  TYMS  UBE2C  UBE2T  VIM  XBP1  XCL1  \
X364.9.0        1  ...     0      0     0      0      0    0     0     0   
X364.11.0       0  ...     0      0     0      0      0    0     0     0   
X364.13.0       0  ...     0      0     0      0      0    0     0     0   
X364.10.001     0  ...     0      0     0      0      0    0     0     0   
X364.12.001     0  ...     0      0     0      0      0    0     0     0   
...           ...  ...   ...    ...   ...    ...    ...  ...   ...   ...   
X364.79.069     0  ...     0      0     0      0      0    0     0     0   
X364.4.071      0  ...     0      0     0      0      0    0     0     0   
X364.8.071      0  ...     0      0     0      0      0    0     0     0   
X364.2.074      0  ...     0      0     0      0      0    0     0     0   
X364.9.074      0  ...     0      0     0      0      0    0     0     0   

             ZEB1  ZNF571  
X364.9.0        0       0  
X364.11.0       0       0  
X364.13.0       1       0  
X364.10.001     0       0  
X364.12.001     0       0  
...           ...     ...  
X364.79.069     0       0  
X364.4.071      0       0  
X364.8.071      0       0  
X364.2.074      0       0  
X364.9.074      0       0  

[447 rows x 287 columns]

In [11]:
df880 = pd.read_csv('counts_1_880.csv', index_col='Unnamed: 0')
df880

ACTA2  ACTG2  ACTR3B  ADGRL4  AGR2  AHR  AIF1  AKT1  ALDH1A3  \
X880.1.003       0      0       0       0     0    0     0     0        0   
X880.2.003       0      0       0       0     0    0     0     0        0   
X880.7.003       0      0       0       0     0    0     0     0        1   
X880.12.003      0      0       0       0     0    0     0     0        0   
X880.13.003      0      0       0       0     0    0     0     0        0   
...            ...    ...     ...     ...   ...  ...   ...   ...      ...   
X880.41.116      0      0       0       0     0    0     0     0        0   
X880.42.116      0      0       0       0     0    0     0     0        0   
X880.47.116      0      0       0       0     0    0     0     0        0   
X880.58.116      0      0       0       0     0    0     0     0        1   
X880.64.116      0      0       0       0     0    0     0     0        0   

             ANLN  ...  TTC6  TTYH1  TYMS  UBE2C  UBE2T  VIM  XBP1  XCL1  \
X880.1.003      0  ...     0      0     0      0      0    1     0     0   
X880.2.003      0  ...     0      0     0      0      0    0     0     0   
X880.7.003      0  ...     0      0     0      0      0    0     0     0   
X880.12.003     0  ...     0      0     0      0      2    0     0     0   
X880.13.003     0  ...     0      0     0      0      0    0     0     0   
...           ...  ...   ...    ...   ...    ...    ...  ...   ...   ...   
X880.41.116     0  ...     0      0     0      1      0    0     0     0   
X880.42.116     0  ...     0      0     0      0      0    0     0     0   
X880.47.116     0  ...     0      0     0      0      0    0     0     0   
X880.58.116     0  ...     0      0     0      0      0    0     0     0   
X880.64.116     0  ...     0      0     0      0      0    0     0     0   

             ZEB1  ZNF571  
X880.1.003      0       0  
X880.2.003      0       0  
X880.7.003      0       1  
X880.12.003     0       0  
X880.13.003     0       0  
...           ...     ...  
X880.41.116     0       0  
X880.42.116     0       0  
X880.47.116     0       0  
X880.58.116     0       0  
X880.64.116     0       0  

[1163 rows x 296 columns]

In [12]:
dftotal = pd.read_csv('combined_counts_1.csv', index_col='cell_name')
dftotal

ACTA2  ACTG2  ACTR3B  ADGRL4  AGR2  AHR  AIF1  AKT1  ALDH1A3  \
cell_name                                                                   
X100.3.001       0      0       1       0   0.0    0     0     0        0   
X100.13.001      0      0       0       0   0.0    0     0     0        0   
X100.21.001      0      0       0       0   0.0    0     0     0        0   
X100.28.001      0      0       0       0   0.0    0     0     0        0   
X100.31.001      0      0       0       0   0.0    0     0     0        0   
...            ...    ...     ...     ...   ...  ...   ...   ...      ...   
X982.15.062      0      0       0       0   0.0    0     0     0        0   
X982.57.062      0      0       0       0   0.0    0     0     0        0   
X982.64.062      1      0       0       0   0.0    0     0     0        0   
X982.67.062      0      0       0       0   0.0    0     0     0        0   
X982.72.062      0      0       0       0   0.0    0     0     0        0   

             ANLN  ...  TYMS  UBE2C  UBE2T  VIM  XBP1  XCL1  ZEB1  ZNF571  \
cell_name          ...                                                      
X100.3.001      1  ...     0      0      0    0   0.0     0     0       1   
X100.13.001     0  ...     0      0      0    0   0.0     0     0       0   
X100.21.001     0  ...     0      0      0    0   0.0     0     0       0   
X100.28.001     0  ...     0      0      0    0   0.0     0     0       0   
X100.31.001     0  ...     0      0      0    0   0.0     0     0       0   
...           ...  ...   ...    ...    ...  ...   ...   ...   ...     ...   
X982.15.062     0  ...     0      0      0    0   0.0     0     0       0   
X982.57.062     0  ...     0      0      0    0   0.0     0     0       0   
X982.64.062     0  ...     0      0      0    0   0.0     0     0       0   
X982.67.062     0  ...     0      0      0    0   0.0     2     0       0   
X982.72.062     0  ...     0      0      0    0   0.0     0     0       0   

             ALB  LIPE  
cell_name               
X100.3.001   0.0   0.0  
X100.13.001  0.0   0.0  
X100.21.001  0.0   0.0  
X100.28.001  0.0   0.0  
X100.31.001  0.0   0.0  
...          ...   ...  
X982.15.062  0.0   0.0  
X982.57.062  0.0   0.0  
X982.64.062  0.0   0.0  
X982.67.062  0.0   0.0  
X982.72.062  0.0   0.0  

[4530 rows x 299 columns]

In [13]:
df982 = pd.read_csv('counts_1_982.csv', index_col='Unnamed: 0')
df982
df982.columns

Index(['ACTA2', 'ACTG2', 'ACTR3B', 'ADGRL4', 'AGR2', 'AHR', 'AIF1', 'AKT1',
       'ALDH1A3', 'ANLN',
       ...
       'TSPAN1', 'TTC6', 'TTYH1', 'TYMS', 'UBE2C', 'UBE2T', 'VIM', 'XCL1',
       'ZEB1', 'ZNF571'],
      dtype='object', length=287)

comper the Gense in each table

In [14]:
Genes_name_100 = df100.columns
Genes_name_313 = df313.columns
Genes_name_364 = df364.columns
Genes_name_880 = df880.columns
Genes_name_982 = df982.columns

 

In [15]:
set_Genes_name_100 = set(Genes_name_100)
set_Genes_name_313 = set(Genes_name_313)
set_Genes_name_364 = set(Genes_name_364)
set_Genes_name_880 = set(Genes_name_880)
set_Genes_name_982 = set(Genes_name_982)

In [16]:
columns_list = [set_Genes_name_100, set_Genes_name_313, set_Genes_name_364, set_Genes_name_880, set_Genes_name_982]
common_columns = set.intersection(*columns_list)

In [17]:
print("Common columns:", common_columns)
len(common_columns)

Common columns: {'KRT18', 'PDPN', 'SNAI2', 'FCRL5', 'IL2RA', 'ALDH1A3', 'PTTG1', 'SKAP1', 'ELF5', 'PHGDH', 'AHR', 'MAPK3', 'SFRP1', 'KRT10', 'HLA-DRB5', 'CD8A', 'SPDEF', 'CDC6', 'TFRC', 'ACTR3B', 'LAMA1', 'FGFR2', 'FXYD3', 'HLA-B', 'IL7R', 'FOXA1', 'GNLY', 'EIF3E', 'SOX4', 'LDB2', 'TCF4', 'CD68', 'FABP7', 'NUF2', 'LYPD6B', 'COL4A2', 'CD36', 'EGFR', 'CR2', 'ESR1', 'S100A8', 'TTYH1', 'COBL', 'SLC2A1', 'NDC80', 'ZEB1', 'PI3', 'AKT1', 'FN1', 'TP53', 'FGFR4', 'ANLN', 'TTC6', 'COL1A1', 'JUN', 'TFF1', 'GRB7', 'MAPK13', 'CD40LG', 'EFNA5', 'UBE2C', 'SIAH2', 'TAGLN', 'LRP2', 'ERBB2', 'SLC39A6', 'BRAF', 'CSTB', 'KRAS', 'PTPRC', 'NF1', 'LGALS1', 'COL1A2', 'FTL', 'PRLR', 'ADGRL4', 'CDKN2A', 'HSPG2', 'THY1', 'DCN', 'FAP', 'KRT7', 'EPCAM', 'GPR160', 'CDK6', 'PABPC1', 'ISG20', 'STAT5A', 'HLA-DPA1', 'COL4A1', 'CCNE2', 'CD44', 'IGF1R', 'CD24', 'CDH1', 'CD163', 'CD5', 'CRABP2', 'CD4', 'THEMIS', 'MUC1', 'KRT15', 'HSPB1', 'LGALS2', 'CCL2', 'CD2', 'APOE', 'AZGP1', 'KRT8', 'GPNMB', 'MYL9', 'MS4A1', 'ORC6', '

272

In [18]:
common_columns_list = list(common_columns)
df100_296 = df100[common_columns_list]
df100_296

KRT18  PDPN  SNAI2  FCRL5  IL2RA  ALDH1A3  PTTG1  SKAP1  ELF5  \
X100.3.001       1     0      0      0      1        0      0      0     0   
X100.13.001      0     0      0      0      0        0      0      0     0   
X100.21.001      0     0      0      0      0        0      0      0     0   
X100.28.001      0     0      0      0      0        0      0      0     0   
X100.31.001      0     0      0      0      0        0      0      0     0   
...            ...   ...    ...    ...    ...      ...    ...    ...   ...   
X100.52.076      0     0      0      0      0        0      0      0     0   
X100.56.076      0     0      0      0      0        0      0      0     0   
X100.57.076      0     0      0      0      0        0      0      0     0   
X100.59.076      0     0      0      0      0        0      0      1     0   
X100.69.076      0     0      0      0      0        0      0      0     0   

             PHGDH  ...  ZNF571  HLA-E  PDCD1  IGHG4  IGKC  CCL3  MYO5B  \
X100.3.001       0  ...       1      0      0      0     0     0      0   
X100.13.001      0  ...       0      0      0      0     0     0      0   
X100.21.001      0  ...       0      0      0      0     0     0      0   
X100.28.001      0  ...       0      0      0      0     0     0      0   
X100.31.001      0  ...       0      0      0      0     0     0      0   
...            ...  ...     ...    ...    ...    ...   ...   ...    ...   
X100.52.076      0  ...       0      0      1      0     0     0      0   
X100.56.076      0  ...       0      0      0      0     0     0      1   
X100.57.076      0  ...       0      0      0      1     0     0      1   
X100.59.076      0  ...       0      0      0      0     0     0      0   
X100.69.076      0  ...       0      0      0      0     0     0      0   

             TMSB10  MYB  CD63  
X100.3.001        0    0     0  
X100.13.001       0    0     0  
X100.21.001       0    0     0  
X100.28.001       0    0     0  
X100.31.001       0    0     0  
...             ...  ...   ...  
X100.52.076       0    0     0  
X100.56.076       0    0     0  
X100.57.076       0    0     0  
X100.59.076       0    0     0  
X100.69.076       0    0     0  

[1545 rows x 272 columns]

In [19]:
df313_296 = df313[common_columns_list]
df313_296

KRT18  PDPN  SNAI2  FCRL5  IL2RA  ALDH1A3  PTTG1  SKAP1  ELF5  \
X313.18.0         0     0      0      0      0        0      0      0     0   
X313.19.0         0     0      0      0      0        0      0      0     0   
X313.22.0         2     0      0      0      0        0      0      0     0   
X313.30.0         1     0      0      0      0        0      0      0     0   
X313.1.001        0     0      0      0      0        0      0      0     0   
...             ...   ...    ...    ...    ...      ...    ...    ...   ...   
X313.85.107       0     0      0      0      0        0      0      0     0   
X313.86.107       0     0      0      0      0        0      0      0     0   
X313.103.107      0     0      0      0      0        0      0      0     0   
X313.105.107      0     0      0      0      0        0      0      0     0   
X313.110.107      0     0      0      0      0        1      0      0     0   

              PHGDH  ...  ZNF571  HLA-E  PDCD1  IGHG4  IGKC  CCL3  MYO5B  \
X313.18.0         1  ...       0      0      0      0     0     0      0   
X313.19.0         0  ...       0      0      0      0     1     0      0   
X313.22.0         3  ...       0      0      0      0     1     0      0   
X313.30.0         1  ...       1      0      0      0     0     0      0   
X313.1.001        0  ...       0      0      0      0     0     0      0   
...             ...  ...     ...    ...    ...    ...   ...   ...    ...   
X313.85.107       0  ...       0      0      0      0     1     0      0   
X313.86.107       0  ...       0      0      0      0     1     0      0   
X313.103.107      0  ...       0      0      0      0     0     0      0   
X313.105.107      0  ...       0      0      0      0     0     0      0   
X313.110.107      0  ...       0      0      0      0     0     0      1   

              TMSB10  MYB  CD63  
X313.18.0          0    0     0  
X313.19.0          0    0     0  
X313.22.0          0    0     0  
X313.30.0          0    0     0  
X313.1.001         0    0     0  
...              ...  ...   ...  
X313.85.107        0    0     3  
X313.86.107        0    0     1  
X313.103.107       0    0     2  
X313.105.107       0    0     0  
X313.110.107       0    0     1  

[645 rows x 272 columns]

In [20]:
df364_296 = df364[common_columns_list]
df364_296

KRT18  PDPN  SNAI2  FCRL5  IL2RA  ALDH1A3  PTTG1  SKAP1  ELF5  \
X364.9.0         0     0      0      0      0        0      0      0     0   
X364.11.0        0     0      0      0      0        0      0      0     0   
X364.13.0        0     0      0      0      0        0      0      0     0   
X364.10.001      3     0      0      0      0        0      0      0     0   
X364.12.001      0     0      0      0      0        0      0      0     0   
...            ...   ...    ...    ...    ...      ...    ...    ...   ...   
X364.79.069      0     0      0      0      0        0      0      0     0   
X364.4.071       0     0      0      0      0        0      0      0     0   
X364.8.071       0     0      0      0      0        0      0      0     0   
X364.2.074       0     0      0      0      0        0      0      0     0   
X364.9.074       0     0      0      0      0        0      0      0     0   

             PHGDH  ...  ZNF571  HLA-E  PDCD1  IGHG4  IGKC  CCL3  MYO5B  \
X364.9.0         1  ...       0      0      0      0     0     0      0   
X364.11.0        0  ...       0      0      0      0     1     0      0   
X364.13.0        2  ...       0      0      0      0     1     1      0   
X364.10.001      3  ...       0      1      0      0     4     0      1   
X364.12.001      0  ...       0      0      0      0     0     0      0   
...            ...  ...     ...    ...    ...    ...   ...   ...    ...   
X364.79.069      0  ...       0      0      0      0     0     0      1   
X364.4.071       0  ...       0      0      0      0     0     0      0   
X364.8.071       3  ...       0      0      0      0     0     0      0   
X364.2.074       1  ...       0      0      0      0     0     0      0   
X364.9.074       0  ...       0      0      0      0     0     0      0   

             TMSB10  MYB  CD63  
X364.9.0          0    0     0  
X364.11.0         0    0     0  
X364.13.0         0    0     0  
X364.10.001       0    0     0  
X364.12.001       0    0     0  
...             ...  ...   ...  
X364.79.069       0    0     0  
X364.4.071        0    0     0  
X364.8.071        0    0     0  
X364.2.074        0    0     0  
X364.9.074        0    0     0  

[447 rows x 272 columns]

In [21]:
df880_296 = df880[common_columns_list]
df880_296

KRT18  PDPN  SNAI2  FCRL5  IL2RA  ALDH1A3  PTTG1  SKAP1  ELF5  \
X880.1.003       0     0      0      0      0        0      0      0     0   
X880.2.003       0     0      0      0      0        0      0      0     0   
X880.7.003       0     1      0      0      0        1      0      0     0   
X880.12.003      0     3      0      0      0        0      0      0     0   
X880.13.003      1     1      0      0      0        0      0      0     0   
...            ...   ...    ...    ...    ...      ...    ...    ...   ...   
X880.41.116      0     0      0      1      0        0      0      0     0   
X880.42.116      0     0      0      0      0        0      0      1     0   
X880.47.116      0     0      0      0      0        0      0      0     0   
X880.58.116      0     0      0      0      0        1      0      0     0   
X880.64.116      0     0      0      0      0        0      0      0     0   

             PHGDH  ...  ZNF571  HLA-E  PDCD1  IGHG4  IGKC  CCL3  MYO5B  \
X880.1.003       0  ...       0      0      0      0     0     0      0   
X880.2.003       0  ...       0      0      0      0     0     0      0   
X880.7.003       0  ...       1      0      0      0     0     0      1   
X880.12.003      0  ...       0      0      0      0     1     0      0   
X880.13.003      0  ...       0      0      0      0     0     0      1   
...            ...  ...     ...    ...    ...    ...   ...   ...    ...   
X880.41.116      0  ...       0      0      0      0     1     0      0   
X880.42.116      0  ...       0      0      0      0     0     1      0   
X880.47.116      0  ...       0      0      0      0     0     0      0   
X880.58.116      0  ...       0      0      0      0     0     0      0   
X880.64.116      0  ...       0      0      0      0     0     0      0   

             TMSB10  MYB  CD63  
X880.1.003        0    0     0  
X880.2.003        0    0     0  
X880.7.003        0    0     0  
X880.12.003       1    0     1  
X880.13.003       0    0     0  
...             ...  ...   ...  
X880.41.116       0    1     0  
X880.42.116       0    0     0  
X880.47.116       0    0     0  
X880.58.116       0    0     0  
X880.64.116       0    0     0  

[1163 rows x 272 columns]

In [22]:
df982_296 = df982[common_columns_list]
df982_296

KRT18  PDPN  SNAI2  FCRL5  IL2RA  ALDH1A3  PTTG1  SKAP1  ELF5  \
X982.10.003      0     0      0      0      0        0      0      0     0   
X982.12.003      0     0      0      0      0        0      0      0     0   
X982.20.003      0     0      0      0      0        0      0      0     0   
X982.4.004       0     0      0      0      0        0      0      0     0   
X982.5.004       0     0      0      0      0        0      0      1     0   
...            ...   ...    ...    ...    ...      ...    ...    ...   ...   
X982.15.062      0     0      0      0      0        0      0      0     0   
X982.57.062      0     0      0      0      0        0      0      0     0   
X982.64.062      0     0      0      0      0        0      0      0     0   
X982.67.062      0     0      0      0      0        0      0      0     0   
X982.72.062      0     0      0      0      0        0      0      0     0   

             PHGDH  ...  ZNF571  HLA-E  PDCD1  IGHG4  IGKC  CCL3  MYO5B  \
X982.10.003      0  ...       0      0      0      0     0     0      0   
X982.12.003      0  ...       0      0      0      0     0     0      1   
X982.20.003      1  ...       0      0      0      0     0     0      0   
X982.4.004       0  ...       0      0      0      0     0     0      0   
X982.5.004       0  ...       0      0      0      0     0     0      0   
...            ...  ...     ...    ...    ...    ...   ...   ...    ...   
X982.15.062      0  ...       0      0      0      0     0     0      0   
X982.57.062      0  ...       0      2      0      0     0     0      2   
X982.64.062      0  ...       0      0      0      0     0     0      3   
X982.67.062      0  ...       0      0      0      0     0     0      0   
X982.72.062      0  ...       0      0      0      0     0     0      0   

             TMSB10  MYB  CD63  
X982.10.003       0    0     0  
X982.12.003       0    0     0  
X982.20.003       0    0     0  
X982.4.004        0    0     0  
X982.5.004        0    1     0  
...             ...  ...   ...  
X982.15.062       0    1     0  
X982.57.062       0    1     0  
X982.64.062       0    0     0  
X982.67.062       0    0     0  
X982.72.062       0    0     0  

[730 rows x 272 columns]

combine the data franes into one of 296 Genes

In [23]:
counts_1_296 = pd.concat([df100_296, df313_296, df364_296, df880_296, df982_296])
counts_1_296.index.name = "cell_name"
counts_1_296

KRT18  PDPN  SNAI2  FCRL5  IL2RA  ALDH1A3  PTTG1  SKAP1  ELF5  \
cell_name                                                                    
X100.3.001       1     0      0      0      1        0      0      0     0   
X100.13.001      0     0      0      0      0        0      0      0     0   
X100.21.001      0     0      0      0      0        0      0      0     0   
X100.28.001      0     0      0      0      0        0      0      0     0   
X100.31.001      0     0      0      0      0        0      0      0     0   
...            ...   ...    ...    ...    ...      ...    ...    ...   ...   
X982.15.062      0     0      0      0      0        0      0      0     0   
X982.57.062      0     0      0      0      0        0      0      0     0   
X982.64.062      0     0      0      0      0        0      0      0     0   
X982.67.062      0     0      0      0      0        0      0      0     0   
X982.72.062      0     0      0      0      0        0      0      0     0   

             PHGDH  ...  ZNF571  HLA-E  PDCD1  IGHG4  IGKC  CCL3  MYO5B  \
cell_name           ...                                                   
X100.3.001       0  ...       1      0      0      0     0     0      0   
X100.13.001      0  ...       0      0      0      0     0     0      0   
X100.21.001      0  ...       0      0      0      0     0     0      0   
X100.28.001      0  ...       0      0      0      0     0     0      0   
X100.31.001      0  ...       0      0      0      0     0     0      0   
...            ...  ...     ...    ...    ...    ...   ...   ...    ...   
X982.15.062      0  ...       0      0      0      0     0     0      0   
X982.57.062      0  ...       0      2      0      0     0     0      2   
X982.64.062      0  ...       0      0      0      0     0     0      3   
X982.67.062      0  ...       0      0      0      0     0     0      0   
X982.72.062      0  ...       0      0      0      0     0     0      0   

             TMSB10  MYB  CD63  
cell_name                       
X100.3.001        0    0     0  
X100.13.001       0    0     0  
X100.21.001       0    0     0  
X100.28.001       0    0     0  
X100.31.001       0    0     0  
...             ...  ...   ...  
X982.15.062       0    1     0  
X982.57.062       0    1     0  
X982.64.062       0    0     0  
X982.67.062       0    0     0  
X982.72.062       0    0     0  

[4530 rows x 272 columns]

In [25]:
counts_1_296.to_csv('conbined_counts_1_272Genes.csv')